docs/debugger_v2.md

```diff
@@ -0,0 +1,303 @@
+[@tf.function](https://www.tensorflow.org/api_docs/python/tf/function),
+the execution of the graphs, the tensor values generated by the execution
+events, as well as the code location (Python stack traces) of those events. The
+richness of the debug information enables users to narrow in on obscure bugs.
+
+```py
+tf.debugging.experimental.enable_dump_debug_info(
+    logdir="/tmp/tfdbg2_logdir",
+    tensor_debug_mode="FULL_HEALTH",
+    circular_buffer_size=-1)
+```
+
+The tensor_debug_mode argument controls what information Debugger V2 extracts
+from each eager or in-graph tensor. “FULL_HEALTH” is a mode that captures the
+following information about each floating-type tensor (e.g., the commonly-seen
+float32 and the less common
+[bfloat16](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) dtype):
+- DType
+- Rank
+- Total number of elements
+-  A breakdown of the floating-type elements into the following categories:
+   negative finite (`-`), zero (`0`), positive finite (`+`), negative infinity
+   (`-∞`), positive infinity (`+∞`), and `NaN`.
+
+The “FULL_HEALTH” mode is suitable for debugging bugs involving NaN and infinity.
+See below for other supported `tensor_debug_mode`s.
+
+The `circular_buffer_size` argument controls how many tensor events are saved to
+the logdir. It defaults to 1000, which causes only the last 1000 tensors before
+the end of the instrumented TF2 program to be saved to disk. This default
+behavior reduces debugger overhead by sacrificing debug-data completeness. If
+the completeness is preferred, as in this case, we can disable the circular
+buffer by setting the argument to a negative value (e.g., -1 here).
+
+The debug_mnist_v2 example invokes `enable_dump_debug_info()` by passing
+command-line flags to it. To run our problematic TF2 program again with this
+debugging instrumentation enabled, do:
+
+```sh
+python -m tensorflow.python.debug.examples.v2.debug_mnist_v2 \
+    --dump_dir /tmp/tfdbg2_logdir --dump_tensor_debug_mode FULL_HEALTH
+```
+
+## Starting the Debugger V2 GUI in TensorBoard
+
+Running the program with the debugger instrumentation creates a logdir at
+/tmp/tfdbg2_logdir. We can point start TensorBoard and point at the logdir by:
+
+```sh
+tensorboard --logdir /tmp/tfdbg2_logdir
+```
+
+In the web browser, navigate to TensorBoard’s page at http://localhost:6006. The
+“Debugger V2” plugin should be activated by default, displaying a page that
+looks like the following:
+
+![Debugger V2 full view screenshot](./images/debugger_v2_1_full_view.png)
+
+## Using Debugger V2 GUI to find the root cause of NaNs
+
+The Debugger V2 GUI in TensorBoard is organized into six sections:
+- **Alerts**: This top-left section contains a list of “alert” events detected
+  by the debugger in the debug data generated by the instrumented TensorFlow
+  program. Each alert indicates a certain anomaly that warrants attention. In
+  our case, this section highlights 442 NaN/∞ events with a salient pink-red
```

Comment:
442?  The images shows 499?

Reply:
Ah - the text was from an older run. It turns out that the number differs slightly depending on whether the TF program runs on CPU or GPU. I made changes here to make the text consistent with the screenshots.